packages/@react-spectrum/s2/src/SegmentedControl.tsx

```diff
@@ -0,0 +1,274 @@
+
+function SegmentedControl(props: SegmentedControlProps, ref: DOMRef<HTMLDivElement>) {
+  [props, ref] = useSpectrumContextProps(props, ref, SegmentedControlContext);
+  let {
+    defaultValue,
+    value
+  } = props;
+  let domRef = useDOMRef(ref);
+
+  let prevRef = useRef<DOMRect>(null);
+  let currentSelectedRef = useRef<HTMLDivElement>(null);
+
+  let onChange = (value: string) => {
+    if (currentSelectedRef.current) {
+      prevRef.current = currentSelectedRef?.current.getBoundingClientRect();
+    }
+    
+    if (props.onChange) {
+      props.onChange(value);
+    }
+  };
+
+  return (
+    <RadioGroup 
+      {...props}
+      ref={domRef}
+      orientation="horizontal"
+      style={props.UNSAFE_style}
+      onChange={onChange}
+      className={(props.UNSAFE_className || '') + segmentedControl({size: 'M'}, props.styles)}
+      aria-label={props['aria-label']}>
+      <DefaultSelectionTracker defaultValue={defaultValue} value={value} prevRef={prevRef} currentSelectedRef={currentSelectedRef}>
+        {props.children}
+      </DefaultSelectionTracker>
+    </RadioGroup>
+  );
+}
+
+function DefaultSelectionTracker(props: DefaultSelectionTrackProps) {
+  let state = useContext(RadioGroupStateContext);
+  let isRegistered = useRef(!(props.defaultValue == null && props.value == null));
+  let disabledIsRegistered = useRef<string | null>(null);
+
+  // default select the first available item
+  let register = useCallback((value: string, isDisabled?: boolean) => {
+    if (state && !isRegistered.current && !isDisabled) {
+      isRegistered.current = true;
+
+      state.setSelectedValue(value);
+    } else if (isDisabled && disabledIsRegistered.current == null) {
+      disabledIsRegistered.current = value;
+    }
+  }, []);
+
+  // if the registration fails, then we will default select the first item
+  useLayoutEffect(() => {
+    if (state && isRegistered.current === false) {
+      console.log(disabledIsRegistered.current);
+      state.setSelectedValue(disabledIsRegistered.current);
+    } else {
+      new Error('Could not determine a default selected item');
+    }
+  }, []);
+
+
+  return (
+    <Provider
+      values={[
+        [InternalSegmentedControlContext, {register: register, prevRef: props.prevRef, currentSelectedRef: props.currentSelectedRef}]
+      ]}> 
+      {props.children}
+    </Provider>
+  );
+}
+
+function ControlItem(props: ControlItemProps, ref: FocusableRef<HTMLLabelElement>) {
+  let inputRef = useRef<HTMLInputElement>(null);
+  let domRef = useFocusableRef(ref, inputRef);
+  let divRef = useRef<HTMLDivElement>(null);
+  let {register, prevRef, currentSelectedRef} = useContext(InternalSegmentedControlContext);
+  let state = useContext(RadioGroupStateContext);
+  let isSelected = props.value === state?.selectedValue;
+  // do not apply animation if a user has the prefers-reduced-motion setting
+  const isReduced = window.matchMedia('(prefers-reduced-motion: reduce)').matches;
```

Comment:
We might need to put this behind an if statement that checks if window?.matchMedia exists otherwise it might not work in environments like server rendering where there is no window